metabase/metabase
# Remove the `field-filter-operators-enabled?` feature flag

https://github.com/metabase/metaboat/issues/144

This PR exposes the operator-based parameter filters to everyone by removing the associated feature flag.

I've also included a small bug fix in https://github.com/metabase/metabase/pull/19378/commits/a3f5e0290af6326cb174d10522e078b2e55ed567. Template tags with widget types of `"category"` and `"location/..."` weren't populating the widget type dropdown in the variables sidebar of the native query editor.

**Testing**
1. Create a native question with template `select * from PRODUCTS where {{category}}` and set the filter type to "field filter" and the widget type to "String". The parameter widget should behave as it would've before if set to a "Category" widget type. The "category" should no longer show up.
2. Create a native question with template `select * from PEOPLE where {{state}}` and set the filter type to "field filter" and the widget type to "Location". Sub location types should no longer be available. Instead, the types should mirror the various string operator types.
3. Create a GUI question (something simple like the Products table) and add it to a dashboard. Create a "Text or Category" > "Dropdown" filter and map it to the Category field of the Products table. Play around with it and make sure it shows field values and filters correctly.
4. Add a Number filter and map it to Products.Rating and make sure it works/filters/etc.
5. Checkout `master` and connect a question on a dashboard to a Category filter. Check this branch out and make sure it still works.
6. Checkout `master` and create a native question with a Category filter and make sure it works after checking this branch back out.
7. Checkout `master` and create a native question with a Location/State filter and make sure it works after checking this branch back out.
